src/analysis/final_attribute.d

```diff
@@ -0,0 +1,335 @@
+//          Copyright Basile Burg 2017.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.final_attribute;
+
+import analysis.base;
+import analysis.helpers;
+import dparse.ast;
+import dparse.lexer;
+
+import std.stdio;
+
+/**
+ * Checks for useless usage of the final attribute.
+ *
+ * There several cases where the compiler allows them even if it's a noop.
+ */
+final class FinalAttributeChecker : BaseAnalyzer
+{
+
+private:
+
+	enum string KEY = "dscanner.useless.final";
+	enum string MSGB = "Useless final attribute, %s";
+
+    static struct MESSAGE
+    {
+        static immutable struct_i    = "structs inherit of anything";
+        static immutable union_i     = "unions inherit of anything";
```

Comment:
The message "Useless final attribute, unions inherit of anything" is hard to understand. Can you think of a way to re-word this?